Create hello-world example

macklinu opened the pull request:
This is a start at #50. The main goal of this PR is to figure out the primary structure for an example directory (README.md structure, how to install `@zeit/ncc`, how to communicate which commands to run, etc.) so that it becomes easier to add other examples moving forward. Feedback welcome (inline comments incoming)!

macklinu commented on examples/hello-world/index.js:
```diff
@@ -0,0 +1 @@
+console.log("Hello World!");
```
This is the simplest Hello World program I could think of, but if there's something else we'd like to express with the Hello World program, I'm happy to change this.

rauchg replied:
Embedding a dependency would be nice. For example, `chalk`

macklinu commented on examples/hello-world/package.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "devDependencies": {
+    "@zeit/ncc": "file:../../"
```
I'm installing ncc from the root of the repo - thoughts?

rauchg replied:
Sounds good!

macklinu replied:
One issue I'm noticing here is that if I run `yarn install` in the `examples/hello-world` directory, it can take ~2 minutes to install `@zeit/ncc` from two directories above.

If I change the version from `file:../../` to `latest` and install from npm, it downloads much faster.

[I notice Next.js' examples use `latest`](https://github.com/zeit/next.js/blob/cd1d3640a9abfa3df3cadf712fe095aa5326a089/examples/using-preact/package.json#L11), and perhaps that's a pattern we could adopt here. Or maybe there's a way to improve the speed of installing a file dependency with yarn using [a command flag](https://yarnpkg.com/lang/en/docs/cli/install/). If you have suggestions about next steps with this, let me know. 😄

rauchg replied:
Let's do `latest` @macklinu. If we install all the dev deps from the directory above, it'll take a very long time. We have all those deps for integration testing purposes only

rauchg replied:
@macklinu ping

macklinu replied:
I addressed this in de48c58, but I don't think GitHub automatically resolved this thread. 😅

macklinu commented on examples/hello-world/README.md:
````diff
@@ -0,0 +1,17 @@
+To execute the Node.js program, run:
+
+```sh
+npx ncc run index.js
````
We could also put these commands in `package.json#scripts`:

### `package.json`

```json
{
  "scripts": {
    "execute": "ncc run index.js",
    "build": "ncc build index.js -o dist"
  }
}
```

### `README.md`

```md
To execute the Node.js program, run `yarn execute`.
To build the Node.js program, run `yarn build`.
```

I chose [npx](https://www.npmjs.com/package/npx) as a way to execute `ncc` from within the `examples/hello-world` directory, but open to feedback on what would be helpful/simple.

rauchg replied:
I like the idea of `scripts` better. `build` and `dev` IMO.

macklinu commented:
@rauchg addressed your comments. Here are examples of running the npm scripts from the `examples/hello-world` directory:

![image](https://user-images.githubusercontent.com/2344137/50036605-e1aed280-ffbe-11e8-90b4-d32bc9a5cf93.png)

![image](https://user-images.githubusercontent.com/2344137/50036611-e8d5e080-ffbe-11e8-8bed-22bcc632e0cb.png)

rauchg commented:
Thanks @macklinu

macklinu commented:
@rauchg this should be a good for another review - apologies for not sending a comment after my latest commit to communicate that more clearly. Let me know if I can help out without anything else on this PR. Will keep contributing example PRs if we like the start of this format. 👍 

(This PR is also good to be squashed and merged unless you'd like me to squash/rebase it manually.)

rauchg commented:
Awesome @macklinu ! Next step would be to show off an example of using `arg` and a CLI with subcommands, where each subcommand is a different bundle
